NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Common/XsqldaMarshaler.cs

```diff
@@ -72,13 +50,14 @@ public void CleanUpNativeData(ref IntPtr pNativeData)
 					IntPtr ptr1 = GetIntPtr(pNativeData, ComputeLength(i));
 
 					// Free	sqldata	and	sqlind pointers	if needed
-					XSQLVAR sqlvar = (XSQLVAR)Marshal.PtrToStructure(ptr1, typeof(XSQLVAR));
+					XSQLVAREmpty sqlvar = (XSQLVAREmpty)Marshal.PtrToStructure(ptr1, typeof(XSQLVAREmpty));
```

Comment:
I took a closer look at all this. Results in latest commit.
I think it is better to have two classes as now. First for manual native-managed marshalling nad managed-native marshalling, and second only for cleanup HGlobal fields. If in this place use XSQLVar Marshal class allocates byte array fields and initializes them.